substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/NativeImageDebugInfoProviderBase.java

```diff
@@ -426,6 +441,25 @@ private static HashMap<JavaKind, HostedType> initJavaKindToHostedTypes(HostedMet
         return map;
     }
 
+    /**
+     * Retrieve details of the native calling convention for a top level compiled method, including
+     * details of which registers or stack slots are used to pass parameters.
+     * 
+     * @param method The method whose calling convention is required.
+     * @return The calling convention for the method.
+     */
+    protected SubstrateCallingConvention getCallingConvention(HostedMethod method) {
+        SubstrateCallingConventionKind callingConventionKind = method.getCallingConventionKind();
+        HostedType declaringClass = method.getDeclaringClass();
+        HostedType receiverType = method.isStatic() ? null : declaringClass;
+        Signature signature = method.getSignature();
+        SubstrateCallingConventionType type = callingConventionKind.toType(false);
+        Backend backend = runtimeConfiguration.lookupBackend(method);
+        RegisterConfig registerConfig = backend.getCodeCache().getRegisterConfig();
+        assert registerConfig instanceof SubstrateRegisterConfig;
+        return (SubstrateCallingConvention) registerConfig.getCallingConvention(type, signature.getReturnType(declaringClass), signature.toParameterTypes(receiverType), backend);
```

Comment:
I'd pass `null` here since `WrappedSignature.getReturnType` doesn't use the `accessingClass` argument anyway.
```suggestion
        return (SubstrateCallingConvention) registerConfig.getCallingConvention(type, signature.getReturnType(null), signature.toParameterTypes(receiverType), backend);
```

Reply:
Done.